src/Head2Head.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from "react";
+import {
+	getRaces,
+	getSlugData,
+	numPagesForGame,
+	RACETIME,
+	SlugData,
+} from "./Loader";
+import FormattedTable from "./FmtTable";
+import CategoryLauncher from "./CategoryLauncher";
+import "./Head2Head.css";
+import MUTable, { MUEntrant, MUMapCtor } from "./MUTable";
+
+// Used to cancel the async function that loads races if a new slug is set.
+let token: Symbol;
+
+function Head2Head() {
+	const [muTable, setMuTable] = useState(new MUTable());
+	const [playerNameList, setPlayerNameList] = useState<Array<string>>([]);
+	const [recordTable, setRecordTable] = useState(MUMapCtor());
+	const [userInfoMap, setUserInfoMap] = useState(new Map<string, MUEntrant>());
+	const [loading, setLoading] = useState(false);
+	const [validSlug, setValidSlug] = useState(false);
+	const [currentSlugData, setCurrentSlugData] = useState<SlugData>();
+	const [loadingProgress, setLoadingProgress] = useState(0);
+	const updateFMTTableProps = (table: MUTable, newPlayersAdded: boolean) => {
+		const [nameList, recTable] = table.generateSortedTable();
+		setRecordTable(recTable);
+		setPlayerNameList(nameList);
+		if (newPlayersAdded) setUserInfoMap(table.playerRaceMap);
+	};
+
+	const setTableSlug = async (slug: string, valid = true) => {
+		setValidSlug(valid);
+		if (!valid) return;
+		const myToken = (token = Symbol());
+		clearTable();
+		getSlugData(slug).then((data) => {
+			setCurrentSlugData(data);
+		});
+		setLoading(true);
+		setLoadingProgress(0);
+		try {
+			const pageLimit = await numPagesForGame(slug);
+			for (let i = 1; i <= pageLimit; i++) {
+				const newTable = muTable;
+				const newPlayersAdded = newTable.processRecordedRaces(
+					(await getRaces(slug, i, true)).races
+				);
+				if (myToken !== token) {
+					clearTable();
+					return;
```

Comment:
Why is the cancelled task messing with the table? The new task already cleared the table, no?